api4/bot_test.go

```diff
@@ -465,7 +465,7 @@ func TestPatchBot(t *testing.T) {
 		CheckCreatedStatus(t, resp)
 		defer th.App.PermanentDeleteBot(createdBot.UserId)
 
-		r, err := th.Client.DoApiPut(th.Client.GetBotRoute(createdBot.UserId), `{"creator_id":"`+th.BasicUser2.Id+`"}`)
+		r, err := th.Client.DoApiPut("/bots/"+createdBot.UserId, `{"creator_id":"`+th.BasicUser2.Id+`"}`)
```

Comment:
Can we use the actual method calls themselves here rather than us constructing the URLs? That seems to be a better solution here.

Same for the rest of the _test.go files.

Reply:
I think the problem is that we are unexporting them so they can't be called from the `api4` package.

Reply:
I meant use the high level methods like - `client.PatchBot`, `client.UpdateConfig` etc. Instead of manually calling Put/Post with the URL.

Reply:
Good suggestion @agnivade :+1: Given that this PR needs to be merged soon I'm incident to deffer these changes to another (HW)-ticket. Does that sound reasonable to you?

Reply:
https://mattermost.atlassian.net/browse/MM-37822

Reply:
@hanzei @agnivade These cases seems like can't use the high level methods provided by client. 

In the above case of patch bot, that pass the `creator_id` to the patch api and client.PatchBot can't do this.

Reply:
I would need to check. Are all cases like that?

Reply:
Yes

Reply:
Thanks for confirming that. In that case we can close the ticket.